src/test/rpc/DeliveredAmount_test.cpp

```diff
@@ -0,0 +1,342 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2019 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/beast/unit_test.h>
+#include <ripple/protocol/Feature.h>
+#include <ripple/protocol/JsonFields.h>
+#include <test/jtx.h>
+#include <test/jtx/WSClient.h>
+
+namespace ripple {
+namespace test {
+
+// Helper class to track the expected number `delivered_amount` results.
+class CheckDeliveredAmount
+{
+    // If the test occurs before or after the switch time
+    bool afterSwitchTime_;
+    // number of payments expected 'delivered_amount' available
+    int numExpectedAvailable_ = 0;
+    // Number of payments with field with `delivered_amount` set to the
+    // string "unavailable"
+    int numExpectedSetUnavailable_ = 0;
+    // Number of payments with no `delivered_amount` field
+    int numExpectedNotSet_ = 0;
+
+    // increment the expected numbers based on xxx, partial payment, and
+    // ter success is true is the ter is tesSUCCESS partial is true if a
+    // partial payment;
```

Comment:
This comment needs a rewrite.

Reply:
Fixed in `Remove empty block and reword comment` commit